rapids-cmake/cython/create_cython_modules.cmake

```diff
@@ -0,0 +1,64 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+#[=======================================================================[.rst:
+create_cython_modules
+---------------------
+
+Generate C(++) from Cython and create Python modules.
+
+.. code-block:: cmake
+
+  create_cython_modules(<ModuleName...>)
+
+Creates a Cython target for a module, then adds a corresponding Python
+extension module. This function assumes that we are in a scikit-build managed
+build and that ``find_package(Cython)`` has already been called.
+
+TODO: Is there any way to avoid the above restriction? I don't think we'd want
+to add a find_package call here, but maybe it would be OK and we could use
+rapids_find_package for tracking.
+
+``cython_modules``
+  The list of modules to build.
+
+``linked_libraries``
+  The list of libraries that need to be linked into all modules. In RAPIDS,
+  this list usually contains (at minimum) the corresponding C++ libraries.
+
+``install_base_directory``
+  The source directory of the project. This directory is used to compute the
+  relative install path, which is necessary to propertly support differently
+  configured installations such as installing in place vs. out of place.
+
+#]=======================================================================]
+function(create_cython_modules cython_modules linked_libraries install_base_directory)
+  foreach(cython_module ${cython_modules})
+    add_cython_target(${cython_module} CXX PY3)
+    add_library(${cython_module} MODULE ${cython_module})
```

Comment:
So am I understanding this correctly, `cython_modules` holds a list of source files and each one is mapped to a new target? 

If that is correct, the style of this should be `rapids_cython_create_modules(MODULES <list of srcs> ... )` instead of having the user specify a variable that holds the sources.

Reply:
I've updated the way that I handle arguments here, let me know if I've done this correctly. I'm still always a little thrown by the way that parameters are handled since anything that isn't named in the function signature just gets passed via argc/argv.